data/provider/src/main/java/tech/pegasys/teku/api/ChainDataProvider.java

```diff
@@ -173,6 +173,20 @@ public boolean isStoreAvailable() {
                             state.hashTreeRoot().toUnprefixedHexString())));
   }
 
+  public SafeFuture<Set<SignedBeaconBlock>> getNonCanonicalBlocks(final String slot) {
+    if (slot.startsWith("0x")) {
+      throw new BadRequestException(
+          String.format("block roots are not currently supported: %s", slot));
+    } else {
+      return defaultBlockSelectorFactory
+          .nonCanonicalBlocksSelector(UInt64.valueOf(slot))
+          .getBlock()
```

Comment:
Shouldn't this be `getBlocks` plural?

Reply:
as discussed, just pre-existing interface.